src/main/java/org/apache/mesos/curator/CuratorStateStore.java

```diff
@@ -137,18 +138,20 @@ public void clearFrameworkId() throws StateStoreException {
     }
 
     @Override
-    public Protos.FrameworkID fetchFrameworkId() throws StateStoreException {
+    public Optional<Protos.FrameworkID> fetchFrameworkId() throws StateStoreException {
         try {
             logger.debug("Fetching FrameworkID from '{}'", fwkIdPath);
             byte[] bytes = curator.fetch(fwkIdPath);
             if (bytes.length > 0) {
-                return Protos.FrameworkID.parseFrom(bytes);
+                return Optional.of(Protos.FrameworkID.parseFrom(bytes));
             } else {
                 throw new StateStoreException(String.format(
                         "Failed to retrieve FrameworkID in '%s'", fwkIdPath));
             }
+        } catch (KeeperException.NoNodeException e) {
+            logger.warn("No FrameworkId found.");
+            return Optional.empty();
```

Comment:
I'm unclear on the what the move to `Optional` buys us here -- in general, I think it's best used when semantics would otherwise dictate returning `null`, rather than as an alternative to exceptional control flow -- but even then, we swallow the one exception here and not the other, which seems to strictly increase the complexity foisted on clients.